:hammer:[refractor][Connverted  Vanilla.js app to be a PWA.]

Ankitv003 opened the pull request:
<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the TBD Developer Website Contributing Guide: https://github.com/TBD54566975/developer.tbd.website/blob/main/CONTRIBUTING.md.
  - 📖 Read the TBD Developer Website Code of Conduct: https://github.com/TBD54566975/developer.tbd.website/blob/main/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [ ] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 👷 Example Application
- [ ] 🎨 Design
- [ ] 📖 Content
- [ ] 🧪 Tests
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

<!-- Please do not leave this blank -->

This PR adds PWA to Vanilla.js App

## Related Tickets & Documents
<!-- 
Please use this format link issue numbers: Resolves #123
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword 
-->
Resolves #85 

## Mobile & Desktop Screenshots/Recordings

<!-- Visual changes require screenshots -->

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### No tests? Add a note
<!-- 
If you didn't provide tests with this PR, please explain here why they aren't needed.
-->

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 general documentation
- [ ] 🙅 no documentation needed

### No docs? Add a note
<!-- 
If you didn't provide documentation with this PR, please explain here why it's not needed.
-->
added comments in code
## [optional] Are there any post-deployment tasks we need to perform?



## [optional] What gif best describes this PR or how it makes you feel?

<!-- Don't know how to add a GIF? 
Here's an example below
![Abbott Elementary GIF](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExNml2MXdzMGU4NzVscnFtbjJzOGJwbmhwYzNrNDJoMjNxMmlsc25uOSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/rEAdbyhgwUSIYrDf8s/giphy.gif)
Other resources for adding GIFs:
- https://docs.google.com/document/d/1pJIVo9NtB5O48t7cKrjhOUACjMacZoAuF38s18QbqnM/edit
- https://stackoverflow.com/questions/34341808/is-there-a-way-to-add-an-animated-gif-to-a-markdown-file
-->



<!-- note: PRs with deletes sections will be marked invalid -->

Ankitv003 commented:
fix issue #85

blackgirlbytes reviewed (changes requested):
Heyy @Ankitv003 !! Thanks for the contribution. Just a few merge conflicts to resolve so we can get this approved and merged. Thank you!!

<img width="804" alt="Screenshot 2024-10-29 at 6 38 16 AM" src="https://github.com/user-attachments/assets/564a8da6-1870-426e-b2ba-12c9fdfa5a74">

Ankitv003 commented on javascript/dwa-starter-vanillajs-vite/tests/main.spec.js:
```diff
@@ -1,26 +1,26 @@
+  // Before all tests, start a local server if necessary
 
-    test('should navigate to Home', async ({ page }) => {
-        await page.goto('/');
```
changed "/" to "http://localhost:5173"

Ankitv003 commented on javascript/dwa-starter-vanillajs-vite/playwright.config.js:
```diff
@@ -101,9 +101,9 @@ export default defineConfig({
      * Use the preview server on CI for more realistic testing.
      * Playwright will re-use the local server if there is already a dev-server running.
      */
-    command: 'pnpm run dev',
```
changed to npm because i am using npm and the testcases were failing

Ankitv003 commented on javascript/dwa-starter-vanillajs-vite/package.json:
```diff
@@ -16,9 +16,13 @@
     "vite": "^5.4.1",
     "vitest": "^2.1.2"
   },
-  "packageManager": "pnpm@9.12.2+sha512.22721b3a11f81661ae1ec68ce1a7b879425a1ca5b991c975b074ac220b187ce56c708fe5db69f4c962c989452eee76c82877f4ee80f474cebd61ee13461b6228"
```
added serwist/next dependency

Ankitv003 commented on javascript/dwa-starter-vanillajs-vite/main.js:
```diff
@@ -1,37 +1,95 @@
+window.addEventListener("popstate", router);
 
 // Initial call to router to render the correct component on page load
-document.addEventListener('DOMContentLoaded', router);
```
registered service worked

Ankitv003 commented on javascript/dwa-starter-vanillajs-vite/index.html:
```diff
@@ -11,7 +11,10 @@
       name="description"
       content="A Decentralized Web Application template"
     />
-    <link rel="stylesheet" href="dist/output.css" />
```
this line is missing only

Ankitv003 commented:
hey @blackgirlbytes  , i have added some comments and i think the only main change required is in index.html file , everything else , either has some extra space or double inverted commas( " ") instead of single(' ').
Also, added some comments on the things i added to the codebase , please look into it , and i am really sorry i didn't mean to spam single comments , this is just my first time making any changes for a PR.

blackgirlbytes commented:
HI @Ankitv003 . Thanks -- Adding comments wont resolve the conflicts though. Here's some information on merge conflicts: 

- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/about-merge-conflicts
-  https://www.reddit.com/r/git/comments/n8yx5h/correct_workflow_to_resolve_conflict_and_merge/
If that's too hard, you may want to pull the latest from main..crerate a new branch..add these same changes and open a new PR. 

I cannot change your code for you because it's on your branch. Thank you!

Ankitv003 commented:
i mean the main change is for index.html , i can change it and then add it for review again , the rest of the files are correct , so you can merge it from your end, right? @blackgirlbytes

blackgirlbytes commented:
@Ankitv003 if you want, you should come to our office hours in [Discord](https://discord.com/events/937858703112155166/1291766030716375103)..so we could help you

Ankitv003 commented:
hey @blackgirlbytes  discord link is not working , what are the office hours ? 👀 I will try to do resolve this by myself today , lets see!

blackgirlbytes commented:
@Ankitv003 hmm okay try discord.gg/tbd and then go to events. You will see our office hours. It's set to happen on Oct 30 at 1pm ET. If you join the discord, you will see it in your timezone

Ankitv003 commented:
made some changes @blackgirlbytes , please check, if it is not ok, i will simply open a new branch,and add a new PR, been on it for most part of the day on a crappy laptop, my head will explode now.

blackgirlbytes commented:
Hey @Ankitv003!!! While I appreciate the updates, I been running the code and it doesn't look like it's working. I'm just seeing a blank page
<img width="1500" alt="Screenshot 2024-10-31 at 3 45 08 PM" src="https://github.com/user-attachments/assets/ea178fb8-26cd-4443-8132-0e770d608305">
<img width="1207" alt="Screenshot 2024-10-31 at 3 42 00 PM" src="https://github.com/user-attachments/assets/7b6fe43e-e89d-48a8-a59a-a6fc8f346731">

Ankitv003 commented:
i think service worker code is missing here , let me check @blackgirlbytes

Ankitv003 commented:
yeah , service worker registeration in main.js was missing and `import { Home, About, Settings, NotFound } from "./components.js";` this line was repeating , i fixed it but how should i add these, i am getting this while sync fork before adding my changes.
![Screenshot (466)](https://github.com/user-attachments/assets/6099ca48-6c89-4bf7-bbb1-933fd00e3dad)
@blackgirlbytes
